src/test/ui/allocator/custom.rs

```diff
@@ -18,7 +18,7 @@ struct A;
 unsafe impl alloc::GlobalAlloc for A {
     unsafe fn alloc(&self, layout: Layout) -> *mut u8 {
         HITS.fetch_add(1, Ordering::SeqCst);
-        System.alloc(layout)
+        AllocRef::alloc(&System, layout).unwrap().as_mut_ptr()
```

Comment:
This should return `ptr::null_mut()` on failure. You can simply use `GlobalAlloc` instead:

```suggestion
        alloc::GlobalAlloc::alloc(&System, layout)
```